src/checkout/context/checkout.tsx

```diff
@@ -119,33 +119,47 @@ export const CheckoutProvider: FC<Props> = React.memo(({children}) => {
   })
 
   const getZuoraParams = useCallback(async () => {
-    const response = await getCheckoutZuoraParams()
-    if (response.status !== 200) {
-      throw new Error(getErrorMessage(response))
-    }
+    try {
+      const response = await getPaymentForm({form: 'checkout'})
+      if (response.status !== 200) {
+        throw new Error(getErrorMessage(response))
+      }
 
-    setZuoraParams(response.data as CreditCardParams)
+      // TODO(ariel): remove type casting here when Billing is refactored and integrated
+      setZuoraParams(response.data as CreditCardParams)
+    } catch (error) {
+      // TODO(ariel): should we notify or report the error?
+      console.error(error)
+    }
   }, [])
 
   useEffect(() => {
     getZuoraParams()
   }, [getZuoraParams])
 
-  const [onSuccessUrl, setOnSuccessUrl] = useState<string>('')
-
   const getBillingSettings = useCallback(
+    // TODO(ariel): this is weird, why do we need a parameter?
     async (fields = inputs) => {
-      const resp = await getBillingNotificationSettings()
-      if (resp.status !== 200) {
-        throw new Error(resp.data.message)
+      try {
+        const resp = await getSettingsNotifications({})
+        if (resp.status !== 200) {
+          throw new Error(resp.data.message)
+        }
+        setInputs({
+          ...fields,
+          balanceThreshold: resp.data.balanceThreshold,
+          notifyEmail: resp.data.notifyEmail,
+          shouldNotify: resp.data.isNotify,
+        })
+      } catch (error) {
+        // We're injesting the error here and leaving the default inputs since
+        // Quartz's API returns a 404 for valid requests that don't have any data
+        // Meaning, if a user hasn't filled out the notification settings, the
+        // API response is expected to return a 404 even though the query was successful
```

Comment:
should we specifically catch 404 errors and only injest those? What if quartz returns a 500 or 403 or something else?

Reply:
also fwiw it seems pretty sketch to return a 404 for a successful response. A 204 seems more appropriate imo

Reply:
Right, I spoke to @Delmer814 and @gspletzer about it yesterday and we kind of came up with the fact that it doesn't matter whether the API returns a successful response or not, the defaults should be set based on existing metadata. I agree that it's weird to set the data in a catch statement and that we should actually handle the `404` specifically, but I think in any situation where data doesn't exist we don't want to override the existing defaults

Reply:
Gotcha, ok that's fair!